cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -144,6 +196,71 @@ protected void runOneIteration() throws Exception {
   }
 
   /**
+   * Perform aggregation on the Streams described by the {@code streamNames}, and no other Streams.
+   * If aggregation was previously done on other Streams, those must be cancelled.
+   *
+   * @param streamNames names of the streams to perform data sizes aggregation on
+   */
+  public void aggregate(Set<String> streamNames) {
+    Set<String> existingAggregators = Sets.newHashSet(aggregators.keySet());
+    for (String streamName : streamNames) {
+      if (existingAggregators.remove(streamName)) {
+        continue;
+      }
+
+      long filesSize = 0;
+      try {
+        StreamConfig config = streamAdmin.getConfig(streamName);
+        filesSize = StreamUtils.fetchStreamFilesSize(config);
+      } catch (IOException e) {
+        LOG.error("Could not compute sizes of files for stream {}", streamName);
+      }
+
+      Aggregator aggregator = new Aggregator(streamName, filesSize);
+      aggregator.startAndWait();
+      aggregators.put(streamName, aggregator);
+    }
+
+    // Stop aggregating the heartbeats we used to listen to before the call to that method,
+    // but don't anymore
+    for (String outdatedStream : existingAggregators) {
+      Aggregator aggregator = aggregators.remove(outdatedStream);
+      aggregator.stopAndWait();
+    }
+  }
+
+  /**
+   * Subscribe to the streams heartbeat notification feed.
+   *
+   * @return a {@link Cancellable} to cancel the subscription
+   * @throws NotificationException if the heartbeat feed does not exist
+   */
+  private Cancellable subscribeToHeartbeatsFeed() throws NotificationException {
+    final NotificationFeed heartbeatsFeed = new NotificationFeed.Builder()
+      .setNamespace("default")
```

Comment:
I will do a pass on all the namespace issues in later PRs. We need to talk about that in more depth. But you're absolutely right to point it out. Here the issue is that we don't subscribe to the heartbeat feed of one stream, but the heartbeat feed is common for all streams. We will certainly have one heartbeat feed per namespace though, and in that case the ``subscribeToHeartbeatsFeed`` would subscribe to all those feeds.